### PR TITLE
chore: rename project to Powertools for AWS Lambda (.NET)

### PR DESCRIPTION
**Pull request by sthulb**
> Please provide the issue number

Issue number: #281 

### Changes

> Please provide a summary of what's being changed

Located all references of Powertools/Lambda Powertools etc and updated to the new project name "Powertools for AWS Lambda"

## Checklist

Please leave checklist items unchecked if they do not apply to your change.

* [X] [Meets tenets criteria](https://awslabs.github.io/aws-lambda-powertools-dotnet/tenets)
* [X] I have performed a self-review of this change
* [X] Changes have been tested
* [X] Changes are documented
* [X] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-dotnet/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: libraries/src/AWS.Lambda.Powertools.Common/Core/Constants.cs [hunk 1]
```diff
@@ -87,7 +87,7 @@ internal static class Constants
     internal const string AwsExecutionEnvironmentVariableName = "AWS_EXECUTION_ENV";
     
     /// <summary>
-    ///     Constant for Powertools feature identifier fo AWS_EXECUTION_ENV environment variable
+    ///     Constant for Powertools for AWS Lambda (.NET) feature identifier fo AWS_EXECUTION_ENV environment variable
```
Comment (hjgraca) [inline libraries/src/AWS.Lambda.Powertools.Common/Core/Constants.cs:90]: I wouldn't change here. comment in code

File: libraries/src/AWS.Lambda.Powertools.Logging/Internal/LoggingAspectHandler.cs [hunk 2]
```diff
@@ -67,7 +67,7 @@ internal class LoggingAspectHandler : IMethodAspectHandler
     private readonly LoggerOutputCase? _loggerOutputCase;
 
     /// <summary>
-    ///     The Powertools configurations
+    ///     The Powertools for AWS Lambda (.NET) configurations
```
Comment (hjgraca) [inline libraries/src/AWS.Lambda.Powertools.Logging/Internal/LoggingAspectHandler.cs:70]: I wouldn't change here. comment in code

File: libraries/src/AWS.Lambda.Powertools.Logging/Internal/PowertoolsConfigurations.cs [hunk 3]
```diff
@@ -27,7 +27,7 @@ internal static class PowertoolsConfigurationsExtension
     /// <summary>
     ///     Gets the log level.
     /// </summary>
-    /// <param name="powertoolsConfigurations">The Powertools configurations.</param>
+    /// <param name="powertoolsConfigurations">The Powertools for AWS Lambda (.NET) configurations.</param>
```
Comment (hjgraca) [inline libraries/src/AWS.Lambda.Powertools.Logging/Internal/PowertoolsConfigurations.cs:30]: I wouldn't change here. comment in code

File: libraries/src/AWS.Lambda.Powertools.Logging/Internal/PowertoolsLogger.cs [hunk 4]
```diff
@@ -65,7 +65,7 @@ internal sealed class PowertoolsLogger : ILogger
     ///     Initializes a new instance of the <see cref="PowertoolsLogger" /> class.
     /// </summary>
     /// <param name="name">The name.</param>
-    /// <param name="powertoolsConfigurations">The Powertools configurations.</param>
+    /// <param name="powertoolsConfigurations">The Powertools for AWS Lambda (.NET) configurations.</param>
```
Comment (hjgraca) [inline libraries/src/AWS.Lambda.Powertools.Logging/Internal/PowertoolsLogger.cs:68]: I wouldn't change here. comment in code

File: libraries/src/AWS.Lambda.Powertools.Metrics/Metrics.cs [hunk 5]
```diff
@@ -37,12 +37,12 @@ public class Metrics : IMetrics
     private readonly MetricsContext _context;
 
     /// <summary>
-    ///     The Powertools configurations
+    ///     The Powertools for AWS Lambda (.NET) configurations
```
Comment (hjgraca) [inline libraries/src/AWS.Lambda.Powertools.Metrics/Metrics.cs:40]: I wouldn't change here. comment in code

**Comment by hjgraca** [comment]
@sthulb we will need to change the dependent repos. for instance [Sam cli](https://github.com/aws/aws-sam-cli-app-templates/tree/master/dotnet6/hello-pt) and aws lambda dotnet [here](https://github.com/aws/aws-lambda-dotnet/tree/master/Blueprints/BlueprintDefinitions/vs2022/PowertoolsFunction) and [here](https://github.com/aws/aws-lambda-dotnet/tree/master/Blueprints/BlueprintDefinitions/vs2022/PowertoolsServerless)

File: libraries/src/AWS.Lambda.Powertools.Logging/Internal/PowertoolsConfigurations.cs (hunk 3 above)
Comment (sthulb) [reply libraries/src/AWS.Lambda.Powertools.Logging/Internal/PowertoolsConfigurations.cs:30]: We're unable to use "Powertools" as a standalone brand any more due to internal issues

File: libraries/src/AWS.Lambda.Powertools.Logging/Internal/LoggingAspectHandler.cs (hunk 2 above)
Comment (sthulb) [reply libraries/src/AWS.Lambda.Powertools.Logging/Internal/LoggingAspectHandler.cs:70]: We're unable to use "Powertools" as a standalone brand any more due to internal issues

File: libraries/src/AWS.Lambda.Powertools.Logging/Internal/PowertoolsLogger.cs (hunk 4 above)
Comment (sthulb) [reply libraries/src/AWS.Lambda.Powertools.Logging/Internal/PowertoolsLogger.cs:68]: We're unable to use "Powertools" as a standalone brand any more due to internal issues

File: libraries/src/AWS.Lambda.Powertools.Metrics/Metrics.cs [hunk 6]
```diff
@@ -37,12 +37,12 @@ public class Metrics : IMetrics
...
     /// </summary>
     private readonly IPowertoolsConfigurations _powertoolsConfigurations;
 
     /// <summary>
-    ///     If true, Powertools will throw an exception on empty metrics when trying to flush
+    ///     If true, Powertools for AWS Lambda (.NET) will throw an exception on empty metrics when trying to flush
```
[the remark this answers is not in the document]
Comment (sthulb) [reply libraries/src/AWS.Lambda.Powertools.Metrics/Metrics.cs:45]: We're unable to use "Powertools" as a standalone brand any more due to internal issues

File: libraries/src/AWS.Lambda.Powertools.Metrics/Metrics.cs [hunk 7]
```diff
@@ -56,7 +56,7 @@ public class Metrics : IMetrics
     ///     format (EMF). See
     ///     https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html
     /// </summary>
-    /// <param name="powertoolsConfigurations">Lambda Powertools Configuration</param>
+    /// <param name="powertoolsConfigurations">Powertools for AWS Lambda (.NET) Configuration</param>
```
[the remark this answers is not in the document]
Comment (sthulb) [reply libraries/src/AWS.Lambda.Powertools.Metrics/Metrics.cs:59]: We're unable to use "Powertools" as a standalone brand any more due to internal issues

File: libraries/src/AWS.Lambda.Powertools.Tracing/Internal/TracingAspectHandler.cs [hunk 8]
```diff
@@ -52,7 +52,7 @@ internal class TracingAspectHandler : IMethodAspectHandler
     private readonly string _namespace;
 
     /// <summary>
-    ///     The Powertools configurations
+    ///     The Powertools for AWS Lambda (.NET) configurations
```
[the remark this answers is not in the document]
Comment (sthulb) [reply libraries/src/AWS.Lambda.Powertools.Tracing/Internal/TracingAspectHandler.cs:55]: We're unable to use "Powertools" as a standalone brand any more due to internal issues

File: libraries/src/AWS.Lambda.Powertools.Tracing/Internal/TracingAspectHandler.cs [hunk 9]
```diff
@@ -77,7 +77,7 @@ internal class TracingAspectHandler : IMethodAspectHandler
     /// <param name="segmentName">Name of the segment.</param>
     /// <param name="nameSpace">The namespace.</param>
     /// <param name="captureMode">The capture mode.</param>
-    /// <param name="powertoolsConfigurations">The Powertools configurations.</param>
+    /// <param name="powertoolsConfigurations">The Powertools for AWS Lambda (.NET) configurations.</param>
```
[the remark this answers is not in the document]
Comment (sthulb) [reply libraries/src/AWS.Lambda.Powertools.Tracing/Internal/TracingAspectHandler.cs:80]: We're unable to use "Powertools" as a standalone brand any more due to internal issues

**Comment by sthulb** [comment]
@hjgraca yep, other repos are on my list to change too 😄 Starting with ours, moving on to other AWS repos, ideally everything will be merged at roughly the same time

**Review by sliedig — APPROVED** [review]

**Pull request merged**